### PR TITLE
PaaSTA instance Envoy replication checks

### PR DESCRIPTION
**Pull request by vkhromov**
In addition to getting replication information from Smartstack, get the
same information from Envoy, and use both of them to send notifications
about underreplicated services.

While here I also cleaned up some `mypy` stuff.

### CODE REVIEW

**Comment by vkhromov** [comment]
Since @gvolco is working on the same files in #2889, I better merge this PR in the current state and then address possible comments in the following PRs.

**Comment by Rob-Johnson** [comment]
big PR, but generally this looks good; just some stylistic changes

**Comment by Rob-Johnson** [comment]
ty for all the cleanup/tidying/formatting work you've done too

File: paasta_tools/envoy_tools.py [hunk 1]
```diff
@@ -100,31 +100,40 @@ def get_backends(
...
     return get_multiple_backends(
-        services, envoy_host=envoy_host, envoy_admin_port=envoy_admin_port,
+        services,
+        envoy_host=envoy_host,
+        envoy_admin_port=envoy_admin_port,
+        envoy_admin_endpoint_format=envoy_admin_endpoint_format,
     )
 
 
 def get_multiple_backends(
```
[the remark this answers is not in the document]
Comment (vkhromov) [reply paasta_tools/envoy_tools.py:110]: The function wasn't introduced in the PR, so let's refactor it separately.

**Pull request merged**